### ISSUE TITLE
VSTest task with code covereage enabled fills up disk with IntelliTrace files

### ISSUE THREAD

**Issue by ChristopherHaws** [Microsoft/vsts-tasks#6694]
## Environment
- Server - VSTS (I can email you account/build details if need be, but I cant post it on a public forum)
- Agent - Private
    - OS: Windows Server 2016 (10.0.14393)
    - Agent Version: 2.129.0

## Issue Description
When using the VSTest Task with CodeCoverage enabled, IntelliTrace logs get placed into the `%WINDIR%\temp\` directory and do not get cleaned up. Our `C:\Windows\Temp` directory was using over 100GB due to `.itrace` files. During the build I was able to locate a file in `F:\vsts\builds\a1\_temp` called `IntelliTrace_Logging_History_15.0_intellitrace_00002bb0_01d3bb24c81e8e1d` with the following content:

```xml
<LoggerInfo xmlns="urn:schemas-microsoft-com:visualstudio:tracelog" name="intellitrace_00002bb0_01d3bb24c81e8e1d" logPath="C:\Windows\Temp\testhost.x86_00002bb0_180313_234117.iTrace" processId="5556" creationTime="0" sessionId="0" keepAlive="false" collectionPlanPath="">
  <ClientInfo id="0" creationTime="0" sessionId="0" name="" processBitness="0" clrVersionMajor="0" commandLine="" />
</LoggerInfo>
```

Here is the yaml of our VSTest task, and we have no custom build variables:
```yaml
steps:
- task: VSTest@2
  inputs:
    testAssemblyVer2: '**\MyProject.Tests.dll'
    vsTestVersion: 15.0
    runInParallel: true
    codeCoverageEnabled: true
    platform: '$(BuildPlatform)'
    configuration: '$(BuildConfiguration)'
    rerunFailedTests: false
```

### Error logs
There are no errors occurring other than disk full errors when the disk runs out of space.

## References
DevCommunity post of this issue: https://developercommunity.visualstudio.com/content/problem/195034/cwindowstemp-fills-up-with-itrace-files-after-upgr.html

**Comment by smadala** [comment]
@ChristopherHaws Thanks for reporting the issue, We have fixed 
 the issue, https://github.com/Microsoft/vstest/pull/1483. Fix will be available in VS 15.7

**Comment by ShreyasRmsft** [comment]
@ChristopherHaws, this has been fixed and is pending release. It should be available around next week. Can't commit to a date yet. You can get the latest vstest platform by using the [test platform tools installer task](https://docs.microsoft.com/en-us/vsts/release-notes/2017/nov-28-vsts#test). 

You can monitor the [nuget package](https://www.nuget.org/packages/Microsoft.TestPlatform/15.7.0-preview-20180307-01) for the new version which should have the fix @smadala made.
The current version is 15.7.0-preview-20180307-01, the next update around next week should have it,

**Comment by ShreyasRmsft** [comment]
Closing this issue now. Please reopen if you need more assistance with the tools installer task or if in the next release this issue still persists.

**Comment by fujiy** [comment]
@smadala, @ShreyasRmsft, this issue also happens when running tests from Visual Studio on the developer machine. It is also fixed in Visual Studio? Or just for TFS Task?

Thanks

**Comment by smadala** [comment]
Yes, Fixed in both.

**Comment by fujiy** [comment]
> @ChristopherHaws, this has been fixed and is pending release. It should be available around next week. Can't commit to a date yet. You can get the latest vstest platform by using the test platform tools installer task.

@ShreyasRmsft, when you say it will be available next week, it will be a RTM version? Or 15.7 Preview 2?

Thanks

**Comment by ShreyasRmsft** [comment]
It will be available for VSTS next week. 

But you'll have to wait longer for the VS RTM release where this will be fix will be released for the IDE as we do not have the convenience to shipping as and when we want. But for VSTS we can and the fix will be out via the nuget package as I have specified above.

**Comment by smadala** [comment]
Fix available in https://www.nuget.org/packages/Microsoft.TestPlatform/15.6.2 . 
We are planning to fix this issue in VS 15.6  upcoming update.

**Comment by NicholasDenaro** [comment]
I am seeing a buildup of .itrace files in VSTest version 16.2.0. I'm using Agent version 2.155.1 x64 on Windows Server 2016.
